### DISCUSSION TITLE
Verwijderen verkeerde data🙋

### DISCUSSION THREAD

**Discussion by FutureCow** [dsmrreader/dsmr-reader#1552] [Questions & answers / How-to's] [answered]
### Description

Ik draai DSMR-Reader nu nog een keer, voor een meter ergens anders. Dit werkt allemaal perfect. Echter heb ik de eerste 2 minuten even mijn eigenmeter uitgelezen, voordat ik de andere meter ging uitlezen. 
Hoe krijg ik die data van mijn meter uit de database?
![P1](https://user-images.githubusercontent.com/2134193/148625832-0f34ac07-969d-41f3-8f7b-5792e7242dc0.png)

### DSMR-reader version

5

### DSMR-reader platform

Docker (e.g. Xirixiz)

**Comment by dennissiemensma** [comment, marked as answer]
Heb je er al veel gegevens in staan? Want je kunt overwegen om anders de database weg te gooien en opnieuw aan te maken en dan de deploy opnieuw te doen.

En anders geldt dit, vanaf morgen zou die dan weer goed moeten gaan: https://dsmr-reader.readthedocs.io/en/latest/how-to/troubleshooting/smart-meter-replacement.html

----

Je kunt handmatig de records verwijderen uit de tabellen `dsmr_datalogger_dsmrreading`, `dsmr_consumption_electricityconsumption` en `dsmr_consumption_gasconsumption` als je de IDs weet.

Bijvoorbeeld:
```
sudo su - postgres
psql dsmrreader

SELECT id, timestamp, electricity_delivered_1 FROM dsmr_datalogger_dsmrreading ORDER BY id ASC LIMIT 100;
# of
SELECT * FROM dsmr_datalogger_dsmrreading ORDER BY id ASC LIMIT 100;
```

En dan kijken wat het eerste ID is die je WEL wilt bewaren. Alles daaronder verwijder je dan:
```
DELETE FROM dsmr_datalogger_dsmrreading WHERE ID < 1234;
```

Voor de andere tabellen kun je hetzelfde doen. Maar gezien het aardig wat werk is, is een schone DB of wachten tot morgen het makkelijkste.

**Comment by FutureCow** [comment]
Bedankt, er zaten 30 verkeerde metingen in. Dus met:

DELETE FROM dsmr_datalogger_dsmrreading WHERE ID < 31;

De verkeerde metingen verwijderd. Welke andere tabellen moet ik dan nog mee pakken?

**Reply by dennissiemensma** [reply]
In mijn vorige comment staan de drie tabellen waar je waarschijnlijk wat uit moet halen. Als de dagtotalen al gegenereerd zijn kun je ze achteraf aanpassen via de admin.

**Reply by dennissiemensma** [reply]
Ik heb nog een verkeerde tabel gekopieerd zie ik, ik zal die aanpassen.

**Comment by FutureCow** [comment]
Top, bedankt!
